### DISCUSSION TITLE
[Projects Beta] Fixed filter vs. user filter

### DISCUSSION THREAD

**Discussion by redcatbear** [community/community#7587] [Projects and Issues]
As a project administrator,
I want to set filters on a per-view basis
so that they define the purpose of the view.

As a project administrator,
I want those filters to be fixed for non-admin users,
so that they can't accidentally destroy the view.

As a project user,
I want to be able narrow down the existing filters by adding more filter criteria,
so that the view still serves its original purpose, but only shows those issues / PRs that are relevant for me.

**Comment by victorlin** [comment]
+1. It is already the non-admin default to have filters in the form of URL parameters, e.g. https://github.com/orgs/github/projects/4247/views/1?filterQuery=repo%3A%22github%2Froadmap%22. So the proposal is to use URL parameters instead of editing the board by default.

**Reply by azenMatt** [reply]
Would I be correct in saying the proposal is to keep that functionality, but have an option (per view?) to 'lock' the settings so the ability to 'save changes' only shows for admins?

**Reply by redcatbear** [reply]
That would definitely better than the current situation.
I would go one step further though. Have a fixed (greyed-out part) of the filter query that comes from the admin and users can _add_ more filter criteria, but not save it. This way the base filter (which gives the view its meaning) is guaranteed to stay intact and regular users can still narrow down the criteria if needed.

**Reply by azenMatt** [reply]
We did explore that! But couldn't quite get it working the way we wanted it. But happy to keep iterating on it 👍 cc @pmarsceill as I know this is a subject you are also interested in 😄

**Reply by victorlin** [reply]
For admins, I propose adding a selection button to choose between viewing and editing mode.

This could also be displayed to non-admins but disabled/fixed to viewing mode.

Like this one in Google Docs:

<img src="https://user-images.githubusercontent.com/13424970/144654378-3b0bf1ed-826e-4eac-ba35-6381f47354a5.png" alt="google docs edit-suggest-view selection" width="300"/>
